parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -44,13 +44,33 @@
 import org.apache.parquet.schema.MessageType;
 
 class ParquetWriter<T> implements FileAppender<T>, Closeable {
-  private static final DynConstructors.Ctor<PageWriteStore> pageStoreCtor = DynConstructors
-      .builder(PageWriteStore.class)
-      .hiddenImpl("org.apache.parquet.hadoop.ColumnChunkPageWriteStore",
-          CodecFactory.BytesCompressor.class,
-          MessageType.class,
-          ByteBufferAllocator.class)
-      .build();
+
+  // We have one for Parquet 1.10 and one for 1.11. The signature changed, but we still want
+  // to be compatible with both of them
+  private static DynConstructors.Ctor<PageWriteStore> pageStoreCtorParquet;
+
+  static {
+    try {
+      // Parquet 1.11
+      pageStoreCtorParquet = DynConstructors
+          .builder(PageWriteStore.class)
+          .hiddenImpl("org.apache.parquet.hadoop.ColumnChunkPageWriteStore",
```

Comment:
These can be added to the same builder to avoid the need for a `static` block like this.

```java
  private static DynConstructors.Ctor<PageWriteStore> ctor = DynConstructors.builder(PageWriteStore.class)
      .hiddenImpl(...) // 1.11
      .hiddenImpl(...) // 1.10
      .build()
```

Reply:
Sweet, that's much cleaner, thanks.

Reply:
Yeah, the purpose of `DynConstructors` and family is to make working with reflection a lot easier. That's why it does things like dropping arguments if there are too many (for exactly this situation) and creating NOOP stand-ins.